### ISSUE TITLE
training problem：no matching

### ISSUE THREAD

**Issue by hfwuyanzu** [cvg/LightGlue#156]
First of all, thank you very much for your outstanding work. I used your training code to try to train a lightglue model with only 6 layers. The training steps follow gluefactory's tutorial exactly. However, the weight file obtained after the end is placed in the matching code, which can extract the key points, but it cannot be matched, and the number of matching pairs is zero. I don't know what the problem is. Please help me.

![Image](https://github.com/user-attachments/assets/534201e2-8291-4673-86de-2b1bdc84081b)
![Image](https://github.com/user-attachments/assets/353608de-e3ff-4aee-a3b3-bd80d71f7333)

Here is my matching code

`from lightglue import LightGlue, SuperPoint, DISK, SIFT, ALIKED, DoGHardNet
from lightglue.utils import load_image, rbd
from lightglue import viz2d
import torch

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")  # 'mps', 'cpu'

# SuperPoint+LightGlue
# extractor = SuperPoint(max_num_keypoints=512).eval().to(device)  # load the extractor
# matcher = LightGlue(features="superpoint",depth_confidence=0.9, width_confidence=0.95).eval().to(device)

#Disk+lightglue
extractor = SuperPoint(max_num_keypoints=512).eval().to(device)  # load the extractor
matcher = LightGlue(features="superpoint").eval().to(device)

# load each image as a torch.Tensor on GPU with shape (3,H,W), normalized in [0,1]
image0 = load_image('/home/jgc/LightGlue/data2/test2/00_src.jpg')
image1 = load_image('/home/jgc/LightGlue/data2/test2/00_tgt.jpg')
# image0 = load_image('assets/sacre_coeur1.jpg')
# image1 = load_image('assets/sacre_coeur2.jpg')

feats0 = extractor.extract(image0.to(device))
feats1 = extractor.extract(image1.to(device))
matches01 = matcher({"image0": feats0, "image1": feats1})

feats0, feats1, matches01 = [
    rbd(x) for x in [feats0, feats1, matches01]
]  # remove batch dimension

kpts0, kpts1, matches = feats0["keypoints"], feats1["keypoints"], matches01["matches"]
m_kpts0, m_kpts1 = kpts0[matches[..., 0]], kpts1[matches[..., 1]]

print(matches.shape)

axes = viz2d.plot_images([image0, image1])
viz2d.plot_matches(m_kpts0, m_kpts1, color="lime", lw=0.2)
viz2d.add_text(0, f'Stop after {matches01["stop"]} layers')

viz2d.save_plot("output1.png")

kpc0, kpc1 = viz2d.cm_prune(matches01["prune0"]), viz2d.cm_prune(matches01["prune1"])
viz2d.plot_images([image0, image1])
viz2d.plot_keypoints([kpts0, kpts1], colors=[kpc0, kpc1], ps=6)

viz2d.save_plot("output2.png")`

**Comment by hfwuyanzu** [comment]
Furthermore, the training results are available on the Hpatches test set
